[framework] Remove the dynamic SE interface in 4.0

valeriosetti opened the pull request:
## Description

It helps resolving https://github.com/Mbed-TLS/mbedtls/issues/8151

## PR checklist

- [ ] **TF-PSA-Crypto PR** 
- [ ] **development PR** https://github.com/Mbed-TLS/mbedtls/pull/10050
- [ ] **3.6 PR** no

gabor-mezei-arm reviewed (dismissed):
LGTM

ronald-cron-arm reviewed (changes requested):
For the sake of 3.6 we should probably do not change anything in the framework, right?

valeriosetti commented:
> For the sake of 3.6 we should probably do not change anything in the framework, right?

Yes, you're absolutely right, sorry. I updated the changes as follows:

- I dropped C code changes because those are guarded by `MBEDTLS_PSA_CRYPTO_SE_C` so they are harmless in 4.0/1.0
- I updated bash test script so that `MBEDTLS_PSA_CRYPTO_SE_C` is `unset` only in the 3.6 branch

ronald-cron-arm reviewed (approved):
LGTM

gabor-mezei-arm reviewed (approved):
LGTM
